ext/ddtrace_profiling_native_extension/collectors_thread_context.c

```diff
@@ -1203,6 +1203,8 @@ void thread_context_collector_sample_allocation(VALUE self_instance, unsigned in
     }
   }
 
+  track_obj_allocation(state->recorder_instance, new_object, sample_weight);
```

Comment:
Minor: More of a naming thing, but `track_obj_allocation` may be a bit confusing during `sample_allocation`. Maybe `track_object`? `heap_recorder_track`? Something like that?